include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -167,60 +177,33 @@ template<class T, class U>
 [[vk::ext_instruction(spv::OpBitcast)]]
 T bitcast(U);
 
-template<class T> // DXC Workaround issue TODO https://github.com/microsoft/DirectXShaderCompiler/issues/6576
-[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
-[[vk::ext_instruction(spv::OpBitcast)]]
-T bitcast(uint64_t);
-
-template<typename Unsigned>
-[[vk::ext_instruction( spv::OpBitFieldUExtract )]]
-Unsigned bitFieldUExtract( Unsigned val, uint32_t offsetBits, uint32_t numBits );
-
-template<typename Signed>
-[[vk::ext_instruction( spv::OpBitFieldSExtract )]]
-Signed bitFieldSExtract( Signed val, uint32_t offsetBits, uint32_t numBits );
-
-}
-
-// should be in `nbl::hlsl::spirv` but would collide right now
-namespace experimental
-{
-using namespace nbl::hlsl::spirv;
-
 template<typename T, typename U>
 [[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
 [[vk::ext_instruction(spv::OpBitcast)]]
-T bitcast(U);
+enable_if_t<is_spirv_type_v<T> && is_spirv_type_v<U>, T> bitcast(U);
 
-template<typename T, uint32_t StorageClass>
+template<typename T>
 [[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
 [[vk::ext_instruction(spv::OpBitcast)]]
-uint64_t bitcast(pointer_t<StorageClass,T>);
+uint64_t bitcast(pointer_t<spv::StorageClassPhysicalStorageBuffer,T>);
 
-template<typename T, uint32_t StorageClass>
+template<typename T>
 [[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
 [[vk::ext_instruction(spv::OpBitcast)]]
-pointer_t<StorageClass,T> bitcast(uint64_t);
+pointer_t<spv::StorageClassPhysicalStorageBuffer,T> bitcast(uint64_t);
 
-template<typename T, typename U, uint32_t StorageClass>
-[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
-[[vk::ext_instruction(spv::OpBitcast)]]
-pointer_t<StorageClass,T> bitcast(pointer_t<StorageClass,U>);
-}
+template<typename Unsigned>
+[[vk::ext_instruction( spv::OpBitFieldUExtract )]]
+Unsigned bitFieldUExtract( Unsigned val, uint32_t offsetBits, uint32_t numBits );
 
-namespace atomic64 {
-template<typename T> // integers operate on 2s complement so same op for signed and unsigned
-[[vk::ext_capability(spv::CapabilityInt64Atomics)]]
-[[vk::ext_instruction(spv::OpAtomicIAdd)]]
-T atomicIAdd([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+template<typename Signed>
+[[vk::ext_instruction( spv::OpBitFieldSExtract )]]
+Signed bitFieldSExtract( Signed val, uint32_t offsetBits, uint32_t numBits );
```

Comment:
you should `enable_if_t<is_unsigned_v<> >` etc. to make these slightly safer to use